app/src/main/java/com/practicum/playlist_maker/SearchActivity.kt

```diff
@@ -1,12 +1,72 @@
 package com.practicum.playlist_maker
 
+import android.content.Context
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.EditText
+import android.widget.ImageView
+import com.practicum.playlist_maker.Сonstants.SEARCH_TEXT
 
 class SearchActivity : AppCompatActivity() {
+    private var searchText = ""
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_search)
 
+        val buttonBack = findViewById<ImageView>(R.id.back)
+        val imageClickListener: View.OnClickListener = View.OnClickListener { finish() }
+        buttonBack.setOnClickListener(imageClickListener)
+
+        val editText = findViewById<EditText>(R.id.edittext_in_search_string)
+        val btnClear = findViewById<ImageView>(R.id.icon_clear_in_string)
+        editText.addTextChangedListener(object : TextWatcher{
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+               // TODO("Not yet implemented")
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+               // TODO("Not yet implemented")
+                if (editText.text.isEmpty()){
+                    btnClear.visibility = View.INVISIBLE
+                }else{
+                    btnClear.visibility = View.VISIBLE
+                }
```

Comment:
Можно сделать проще:
```suggestion
         btnClear.isVisible = editText.text.isNotEmpty()
```

Reply:
Поправил